index.js

```diff
@@ -1,39 +1,138 @@
+  c: '&search_scope=standard_number',
+}
+
+const getIndexMapping = index => indexMappings[index] || '';
+
+const recodeSearchQuery = query => query.split(/\+|\s/).join("%20");
+
+const getQueryFromParams = (url, query) => {
+  const getParam = param => (query[param] || [])[0];
+  const searchArg = getParam('searcharg');
+  const searchArgAlt = getParam('SEARCH');
+  let searchIndex;
+  let searchArgFromQueryParam;
+  for (let key in query) {
+    const value = getParam(key);
+    if (!value) {
+      const splitted = key.split('/');
+        if (splitted[1]) {
+          searchIndex = splitted[1][0];
+          searchArgFromQueryParam = splitted[1].slice(1);
+          break;
+        }
+        else if (splitted[0]) {
+          searchArgFromQueryParam = splitted[0];
+          break;
+        }
+    }
+  }
+  const q = searchArg || searchArgAlt || searchArgFromQueryParam || '';
+  const searchType = getParam('searchtype') || searchIndex;
+  return recodeSearchQuery(q + (searchType ? getIndexMapping(searchType) : ''));
+}
 
-// the main mapping method
+// declare some regular expressions
 
-const mapWebPacUrlToSCCURL = (url) => {
+const expressions = {
+  nothingReg: {
+    expr: /^\/$/,
+    handler: () => BASE_SCC_URL,
+  },
+  searchRegWith: {
+    expr: /\/search(~S\d*)?\/([a-zA-Z])(([^\/])+)/,
+    handler: match => `${BASE_SCC_URL}/search?q=${recodeSearchQuery(match[3])}${getIndexMapping(match[2])}`
+  },
+  searchRegWithout: {
+    expr: /\/search(~S\d*)?(\/([a-zA-Z]))?/,
+    handler: (match, query) => `${BASE_SCC_URL}/search?q=${getQueryFromParams(match[0], query)}${getIndexMapping(match[3])}`
+  },
+  patroninfoReg: {
+    expr: /\/patroninfo[^\/]\/(\d+)/,
+    handler: match => `${BASE_SCC_URL}?fakepatron=${match[1]}`
+  },
+  recordReg: {
+    expr: /\/record=(\w+)/,
+    handler: (match) => {
+      const bnum = match[1];
+      return `${BASE_SCC_URL}/bib/${bnum}`;
+    }
+  },
+};
+
+async function mapWebPacUrlToSCCURL(path, query) {
+  console.log('mapping');
   let redirectURL;
-  const parsedURL = new URL(url);
-  const pathname = parsedURL.pathname
-  for (let pathType of expressions) {
-      const match = pathname.match(pathType.expr);
+  for (let pathType of Object.values(expressions)) {
+      const match = path.match(pathType.expr);
       if (match) {
-        redirectURL = pathType.handler(match, parsedURL);
+        console.log('matching: ', pathType);
+        redirectURL = await pathType.handler(match, query);
         break
       }
   }
+  if (!redirectURL) redirectURL = BASE_SCC_URL;
+  console.log('returning: ', redirectURL);
   return redirectURL;
 }
 
 const handler = async (event, context, callback) => {
-  const redirectLocation = mapWebPacUrlToSCCURL(event.queryStringParameters.origin);
-  const response = redirectLocation ?
-    {
+  try {
+    console.log('env vars: ', process.env);
+    console.log('event: ', event.path, event.multiValueQueryStringParameters);
```

Comment:
Some stray console.logs in here

Reply:
I thought it might be useful to leave these in for a little bit, until QA is finished.